RFS/src/main/java/org/opensearch/migrations/bulkload/worker/IndexRunner.java

```diff
@@ -33,44 +34,56 @@ public IndexMetadataResults migrateIndices(MigrationMode mode, ICreateIndexConte
         repoDataProvider.getIndicesInSnapshot(snapshotName)
             .stream()
             .forEach(index -> {
-                CreationResult creationResult;
+                List<CreationResult> creationResults;
                 if (skipCreation.test(index.getName())) {
                     log.atInfo()
                         .setMessage("Index {} was not part of the allowlist and will not be migrated.")
                         .addArgument(index.getName())
                         .log();
-                    creationResult = CreationResult.builder()
+                    creationResults = List.of(CreationResult.builder()
                         .name(index.getName())
                         .failureType(CreationFailureType.SKIPPED_DUE_TO_FILTER)
-                        .build();
+                        .build());
                 } else {
-                    creationResult = createIndex(index.getName(), mode, context);
+                    creationResults = createIndex(index.getName(), mode, context);
                 }
 
-                results.index(creationResult);
+                creationResults.forEach(results::index);
 
                 var indexMetadata = metadataFactory.fromRepo(snapshotName, index.getName());
                 indexMetadata.getAliases().fieldNames().forEachRemaining(alias -> {
                     var aliasResult = CreationResult.builder().name(alias);
-                    aliasResult.failureType(creationResult.getFailureType());
+                    aliasResult.failureType(creationResults.get(0).getFailureType());
                     results.alias(aliasResult.build());
                 });
             });
         return results.build();
     }
 
-    private CreationResult createIndex(String indexName, MigrationMode mode, ICreateIndexContext context) {
+    private List<CreationResult> createIndex(String indexName, MigrationMode mode, ICreateIndexContext context) {
         var originalIndexMetadata = metadataFactory.fromRepo(snapshotName, indexName);
         var indexMetadata = originalIndexMetadata.deepCopy();
+        List<CreationResult> creationResults = new ArrayList<>();
         try {
-            indexMetadata = transformer.transformIndexMetadata(indexMetadata);
-            return indexCreator.create(indexMetadata, mode, context);
+            List<IndexMetadata> transformedMetadataList = transformer.transformIndexMetadata(indexMetadata);
```

Comment:
Agree w/ the SonarQube warning - if this fails, just return a List.of(...).
Then, if it succeeds, go into the for loop w/ the separate try-catch blocks